packages/dev/core/src/States/alphaCullingState.ts

```diff
@@ -101,36 +109,70 @@ export class AlphaState {
         this._isBlendConstantsDirty = false;
     }
 
-    public apply(gl: WebGLRenderingContext) {
+    public apply(gl: WebGLRenderingContext, numTargets: number = 1): void {
         if (!this.isDirty) {
             return;
         }
 
         // Alpha blend
         if (this._isAlphaBlendDirty) {
-            if (this._alphaBlend) {
-                gl.enable(gl.BLEND);
+            if (numTargets === 1) {
+                if (this._alphaBlend[0]) {
+                    gl.enable(gl.BLEND);
+                } else {
+                    gl.disable(gl.BLEND);
+                }
             } else {
-                gl.disable(gl.BLEND);
+                const gl2 = gl as WebGL2RenderingContext;

```

Comment:
isnt this dangerous ?

Reply:
The OES_draw_buffers_indexed extension (which allows different blending parameters per target) is only supported in WebGL2, so I think it's safe to do the cast here.